src/Utils/samples.jl

```diff
@@ -209,6 +235,31 @@ function _sample!(D::Vector{VN},
     return D
 end
 
+function _sample!(D::Vector{VN},
+                  sampler::PolytopeSampler{<:LazySet, <:DefaultUniform};
+                  rng::AbstractRNG=GLOBAL_RNG,
+                  seed::Union{Int, Nothing}=nothing
+                 ) where {N, VN<:AbstractVector{N}}
+    rng = reseed(rng, seed)
+    U = DefaultUniform(zero(N), one(N))
+    P = set(sampler)
+    vlist = vertices_list(P)
+    m = length(vlist)
+
+    @inbounds for i in 1:length(D)
```

Comment:
The call to `rand` is not using `U` and the given `rng`. I propose the following loop. I made some tests and the runtime does not improve, but it allocate a factor of ~2x less because of the in-place updates of the random number generator. That is good because it handles better the case when the number of samples is very large.

I defined two new functions that dispatch on the uniform sampler; the first one is not used (but it is useful in general, let's add it), and the second one is the in-place version.

```julia
function Base.rand(rng::AbstractRNG, U::DefaultUniform, n::Int)
    [rand(rng, U) for i in 1:n]
end

function rand!(x, rng::AbstractRNG, U::DefaultUniform)
    @inbounds for i in eachindex(x)
        x[i] = rand(rng, U)
    end
end

function _sample!(....) 
   . . . .

    # combination coefficients
    r = Vector{N}(undef, m-1)

    @inbounds for i in 1:length(D)

        # get a list of m uniform numbers (https://cs.stackexchange.com/a/3229)
        # and compute the corresponding linear combination in-place
        rand!(r, rng, U)
        sort!(r)
        D[i] = r[1] * vlist[1]
        for j in 2:m-1
            α = r[j] - r[j-1]
            D[i] .+= α * vlist[j]
        end
        D[i] .+= (1 - r[m-1]) * vlist[m]
    end
    return D
end
```

Reply:
Thanks, I added your suggestions.